### PR TITLE
Introduce nodenetworkconfigurationenactment_controller

### PR DESCRIPTION
**Pull request by AlonaKaplan**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind enhancement

**What this PR does / why we need it**:
The purpose of this controller is to clean up enactments their policy was already deleted but for some reason the policy controller missed the policy removal event (for example the node was down when the policy was removed).

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: controllers/nodenetworkconfigurationenactment_controller.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		log.Error(err, "Error retrieving policy")
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{RequeueAfter: enactment.EnactmentRefreshWithJitter()}, nil
+}
+
+func (r *NodeNetworkConfigurationEnactmentReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// By default all this functors return true so controller watch all events,
+	// but we only want to watch create/delete for current node.
```
Comment (rhrazdil) [inline controllers/nodenetworkconfigurationenactment_controller.go:87]: `onCreationForThisEnactment` returns `true` only for CreateEvent, I think we are only interested in create event, right?

File: controllers/nodenetworkconfigurationenactment_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		BeforeEach(func() {
+			request.Name = enactment.Name
+		})
+		It("should re-enqueue", func() {
+			result, err := reconciler.Reconcile(context.Background(), request)
+			Expect(err).ToNot(HaveOccurred())
+			expectRequeueAfterIsSetWithEnactmentRefresh(result)
+		})
+	})
+	Context("and policy doen't exist", func() {
```
Comment (rhrazdil) [inline controllers/nodenetworkconfigurationenactment_controller_test.go:78]: nit: doesn't

File: pkg/enactment/refresh.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+)
+
+const (
+	EnactmentRefresh          = 5 * time.Hour
+	EnactmentRefreshMaxFactor = 0.1
+)
+
+// EnactmentRefreshWithJitter add some jitter to to the refresh rate so it does
+// not hit apiserver at the same time.
+func EnactmentRefreshWithJitter() time.Duration {
```
Comment (rhrazdil) [inline pkg/enactment/refresh.go:16]: Please consider naming this function just `EnactmentRefreshWithJitter`, that way this func would be called as `enactment.RefreshRateWithJitter()`, which reads better

File: pkg/enactment/refresh.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	EnactmentRefresh          = 5 * time.Hour
+	EnactmentRefreshMaxFactor = 0.1
+)
+
+// EnactmentRefreshWithJitter add some jitter to to the refresh rate so it does
```
Comment (rhrazdil) [inline pkg/enactment/refresh.go:14]: nit: ...adds ~~some~~ jitter to ~~to~~....

**Review by rhrazdil — CHANGES_REQUESTED** [review]
Nice, I have only a few nits and one suggestion regarding func name.

One concern I thought about is upgrade compatibility with old NNCEs, which are missing the new label.
Since this PR is a solution for corner case, and not primary mechanism for NNCE cleanup, so I believe we're good.

File: pkg/enactment/refresh.go (hunk 3 above)
Comment (AlonaKaplan) [reply pkg/enactment/refresh.go]: done

**Comment by AlonaKaplan** [comment]
> Nice, I have only a few nits and one suggestion regarding func name.
> 
> One concern I thought about is upgrade compatibility with old NNCEs, which are missing the new label.
> Since this PR is a solution for corner case, and not primary mechanism for NNCE cleanup, so I believe we're good.

You're right, the old NNCEs will miss the label. But the old NNCEs have the policy as their owner ref, so we should be covered.

File: controllers/nodenetworkconfigurationenactment_controller.go (hunk 1 above)
Comment (rhrazdil) [reply controllers/nodenetworkconfigurationenactment_controller.go:87]: Thanks, I meant to point out the the comment says 
>// but we only want to watch create/delete for current node.

Which is inconsistent with the code

**Comment by rhrazdil** [comment]
> You're right, the old NNCEs will miss the label. But the old NNCEs have the policy as their owner ref, so we should be covered.

Ahh of course! We're all good then

**Review by rhrazdil — APPROVED** [review]
Thanks!
/approve
/lgtm

**Pull request merged**